core/src/main/java/org/testcontainers/containers/Network.java

```diff
@@ -52,7 +52,7 @@ static Network newNetwork() {
         private final AtomicBoolean initialized = new AtomicBoolean();
 
         @Override
-        public String getId() {
+        public synchronized String getId() {
```

Comment:
Nasty bug discovered during my experiments with Kafka module, concurrent access was giving `null` for the second invocation

Reply:
Have you tried making `id` volatile. This should work and is cheaper under contention that `synchronized` method

Reply:
not a performance critical part (being called just a few times), doesn't make sense to over-optimize it :)

Reply:
Train your good habits! :)

Reply:
keep it simple 😎

Reply:
Isn't `volatile` as simple? Doesn't really matter for me in this case :grin: